plugins/flytekit-identity-aware-proxy/flytekitplugins/identity_aware_proxy/cli.py

```diff
@@ -0,0 +1,167 @@
+                endpoint=self.audience,
+                # See step 3 in https://cloud.google.com/iap/docs/authentication-howto#signing_in_to_the_application
+                auth_endpoint="https://accounts.google.com/o/oauth2/v2/auth",
+                token_endpoint="https://oauth2.googleapis.com/token",
+                # See step 3 in https://cloud.google.com/iap/docs/authentication-howto#signing_in_to_the_application
+                scopes=["openid", "email"],
+                client_id=self.client_id,
+                redirect_uri=self.redirect_uri,
+                verify=self._verify,
+                # See step 3 in https://cloud.google.com/iap/docs/authentication-howto#signing_in_to_the_application
+                request_auth_code_params={
+                    "cred_ref": "true",
+                    "access_type": "offline",
+                },
+                # See step 4 in https://cloud.google.com/iap/docs/authentication-howto#signing_in_to_the_application
+                request_access_token_params={
+                    "client_id": self.client_id,
+                    "client_secret": self.client_secret,
+                    "audience": self.audience,
+                    "redirect_uri": self.redirect_uri,
+                },
+                # See https://cloud.google.com/iap/docs/authentication-howto#refresh_token
+                refresh_access_token_params={
+                    "client_secret": self.client_secret,
+                    "audience": self.audience,
+                },
+            )
+
+    def refresh_credentials(self):
+        """Refresh the IAP credentials. If no credentials are found, it will kick off a full OAuth 2.0 authorization flow."""
+        self._initialize_auth_client()
+        if self._creds:
+            """We have an id token so lets try to refresh it"""
+            try:
+                self._creds = self._auth_client.refresh_access_token(self._creds)
+                if self._creds:
+                    KeyringStore.store(self._creds)
+                return
+            except AccessTokenNotFoundError:
+                logging.warning("Failed to refresh token. Kicking off a full authorization flow.")
+                KeyringStore.delete(self._endpoint)
+
+        self._creds = self._auth_client.get_creds_from_remote()
+        KeyringStore.store(self._creds)
+
+
+def get_gcp_secret_manager_secret(project_id: str, secret_id: str, version: typing.Optional[str] = "latest"):
+    """Retrieve secret from GCP secret manager."""
+    client = secretmanager.SecretManagerServiceClient()
+    name = f"projects/{project_id}/secrets/{secret_id}/versions/{version}"
+    try:
+        response = client.access_secret_version(name=name)
+    except NotFound as e:
+        raise click.BadParameter(e.message)
+    payload = response.payload.data.decode("UTF-8")
+    return payload
+
+
+@click.group()
+def cli():
+    """Generate ID tokens for GCP Identity Aware Proxy (IAP)."""
+    pass
```

Comment:
Having a command group, even with only 1 subcommand currently, could allow to add a command to generate id tokens for service accounts in the future by adding:

```python
@cli.command()
def generate_sa_id_token():
    """Generate a service account ID token for proxy-authentication/authorization with GCP Identity Aware Proxy."""
    raise NotImplementedError
```